### PR TITLE
Issue #5960: Environment variables ignored if the value provided in the user configuration file.

### PR DESCRIPTION
**Pull request by sonnykt**
Github issue: https://github.com/drush-ops/drush/issues/5960

Current behaviour:
Screenshot 2024-04-19 at 2 30 58 PM

Fixed behaviour:
Screenshot 2024-04-19 at 2 35 56 PM

### CODE REVIEW

**Review by greg-1-anderson — APPROVED** [review]
LGTM

File: tests/unit/ConfigLocatorTest.php [hunk 1]
```diff
@@ -91,6 +91,24 @@ function ($item) {
         $this->assertEquals($expected, $aliasPaths);
     }
 
+    /**
+     * Test an env var supersedes user's configuration file.
+     */
+    public function testEnvVar()
+    {
+        putenv('TEST_HOME=An env overridden setting');
```
Comment (weitzman) [inline tests/unit/ConfigLocatorTest.php]: How about making sure that we get user value initially, and that it changes after the env var is set.

Comment (sonnykt) [reply tests/unit/ConfigLocatorTest.php]: Isn't the existing test `testLoadAll()` already checking that?

Comment (sonnykt) [reply tests/unit/ConfigLocatorTest.php]: Anyway, I've updated the test.

**Comment by AlexSkrypnyk** [comment]
@weitzman 
Could you please let me know if there anything else that needs to be done here to help merging this. Thank

**Comment by weitzman** [comment]
Merged ... Even though this is a bug fix, I'm hesitant to backport it as some sites may be relying on the buggy behavior.

**Pull request merged**